website/cue/reference/components/sinks/file.cue

```diff
@@ -20,7 +20,7 @@ components: sinks: file: {
 			compression: {
 				enabled: true
 				default: "none"
-				algorithms: ["none", "gzip"]
+				algorithms: ["none", "gzip", "zstd"]
 				levels: ["none", "fast", "default", "best", 0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
```

Comment:
By the way, could someone explain me, where in the Vector documentation these compression levels are described? I cannot find related information [here](https://vector.dev/docs/reference/configuration/sinks/file/#compression). Is it rendered at all?

Reply:
Yeah, it looks like that isn't currently being rendered correctly on the website. The `file` source also doesn't appear to actually allow for configuration of `levels` (though other places `compression` is used does).

Reply:
So should we do with a compression level? Fix them in a separate PR?

Reply:
i suppose so, both for rendering & how to match them with available algos

Reply:
I think we can just drop `levels` here since it isn't actually used by the `file` sink. When we add support for compression levels there, we can re-add it.

Reply:
So in this case I guess we need to at least specify, which default level is used for each compression algorithm (since Vector uses some defaults). It could be kinda surprising for the users, if the default level is 0 (no or almost no compression).

Reply:
Yeah, that's fair. You can update the algorithm descriptions here:

https://github.com/vectordotdev/vector/blob/553763f6ed37f1bfda7655f9aa6679e2423d27f9/website/cue/reference/components/sinks.cue#L146-L162

To include their default levels.

Reply:
ah right, I added a generic note about Default level being used, but i will add numeric equivalent for each algo too. i will then check the cue whitespace issue

Reply:
oops, the doc here turns out important for zstd too. [flate2 Default is `6`](https://docs.rs/flate2/1.0.24/src/flate2/lib.rs.html#194) which is sensible.

However, async_compression set [zstd `Default` as `0`](https://docs.rs/async-compression/0.3.12/src/async_compression/lib.rs.html#247), rather than the [libzstd value of `3`](https://docs.rs/zstd/latest/zstd/constant.DEFAULT_COMPRESSION_LEVEL.html).

I think we can suggest `async-compression` to use public constant, and then update vector documentation later.

Reply:
I suggest document the current default value right now. When `async-compression` will update their default compression level, we will update Vector documentation too.

Reasoning is simple: we do not know, when `async-compression` will update their code.